src/backend/table_builder.rs

```diff
@@ -46,6 +46,14 @@ pub trait TableBuilder:
             count += 1;
         }
 
+        for check in create.check.iter() {
```

Comment:
@billy1624 maybe is better use `create.check.iter().fold()`, like in other places in `src/backend`?

Reply:
Good call. Please check https://github.com/SeaQL/sea-query/pull/567/commits/8b9a1d93bc2f0a41e78d5fa9c4112e85a1f863f5. I rewrite the whole part.

Reply:
Hmmm, I mean:
```
.fold(true, |first, _| {
    ...
    false
}
```

Reply:
I like the the `for_each` version over the `fold` version

- `for_each`:
    ```rs
    let mut first = true;

    create.columns.iter().for_each(|column_def| {
        if !first {
            write!(sql, ", ").unwrap();
        }
        self.prepare_column_def(column_def, sql);
        first = false;
    });

    create.indexes.iter().for_each(|index| {
        if !first {
            write!(sql, ", ").unwrap();
        }
        self.prepare_table_index_expression(index, sql);
        first = false;
    });
    ```
- `fold`:
    ```rs
    let mut first = true;

    first = create.columns.iter().fold(first, |first, column_def| {
        if !first {
            write!(sql, ", ").unwrap();
        }
        self.prepare_column_def(column_def, sql);
        false
    });

    first = create.indexes.iter().fold(first, |first, index| {
        if !first {
            write!(sql, ", ").unwrap();
        }
        self.prepare_table_index_expression(index, sql);
        false
    });
    ```

Reply:
@billy1624 thank you for explanation, yes, it looks better.